store/iavl/store.go

```diff
@@ -29,18 +31,38 @@ var (
 
 // Store Implements types.KVStore and CommitKVStore.
 type Store struct {
-	tree Tree
+	tree    Tree
+	pruning types.PruningOptions
 }
 
-// LoadStore returns an IAVL Store as a CommitKVStore. Internally it will load the
+// LoadStore returns an IAVL Store as a CommitKVStore. Internally, it will load the
 // store's version (id) from the provided DB. An error is returned if the version
 // fails to load.
 func LoadStore(db dbm.DB, id types.CommitID, pruning types.PruningOptions, lazyLoading bool) (types.CommitKVStore, error) {
+	if !pruning.IsValid() {
+		return nil, fmt.Errorf("pruning options are invalid: %v", pruning)
+	}
+
+	var keepRecent int64
+
+	// Determine the value of keepRecent based on the following:
+	//
+	// If KeepEvery = 1, keepRecent should be 0 since there is no need to keep
+	// latest version in a in-memory cache.
+	//
+	// If KeepEvery > 1, keepRecent should be 1 so that state changes in between
```

Comment:
Huh, that's what 1 does?
I would have though it would have to equal `KeepEvery` or something.
But I guess it doesn't if we only want "query last block"